### PR TITLE
Use printf in validate.sh for proper coloring

### PR DESCRIPTION
**Pull request by jasagredo**
See https://www.shellcheck.net/wiki/SC2028

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).

### CODE REVIEW

File: validate.sh [hunk 1]
```diff
@@ -124,7 +131,7 @@ timed() {
 print_header() {
     TITLE=$1
     TITLEPAT="$(echo "$TITLE"|sed 's:.:=:g')"
-    echo "$CYAN===X============================================================ $(date +%T) ===$RESET" \
+    cyan "===X========================================================================== $(date +%T) ===" \
```
Comment (jasagredo) [inline validate.sh:134]: `Step Build: full build plan (cached and to-be-built dependencies):` didn't fit the previous length.

File: validate.sh [hunk 2]
```diff
@@ -287,15 +294,19 @@ TESTSUITEJOBS="-j$JOBS"
...
-if [ "$OSTYPE" = "msys" ]; then
-    ARCH="x86_64-windows"
-elif [ "$(uname)" = "Linux" ]; then
-    ARCH="x86_64-linux"
-else
-    ARCH="x86_64-osx"
-fi
+RUNHASKELL=$(echo "$HC" | sed -E 's/ghc(-[0-9.]*)$/runghc\1/')
+
+case "$(uname)" in
```
Comment (jasagredo) [inline validate.sh:299]: `uname` is available in mingw. also `OSTYPE` is not defined in posix.

**Review by ulysses4ever — APPROVED** [review]
Cool!

It’s a little funny that red and green have two arguments and blue and cyan have only one. But oh well…

**Review by geekosaur — APPROVED** [review]

**Comment by jasagredo** [comment]
From @ulysses4ever 
> Cool!
> 
> It’s a little funny that red and green have two arguments and blue and cyan have only one. But oh well…

Indeed. Red and green also output the reported time in white, but blue and cyan are only used to report starting of commands. I could put two arguments on all but then call blue and cyan with `""` as second argument.

**Comment by ulysses4ever** [comment]
@jasagredo I think there's no need in extra arguments but rather there's a need in a comment explaining exactly what you just said about the existing arguments

**Comment by jasagredo** [comment]
I guess this just needs a rebase on top of `master`. Will do later today hopefully.

**Pull request merged**